src/engine/Match_extract_mode.ml

```diff
@@ -126,19 +126,53 @@ let offsets_of_mval extract_mvalue =
            end_pos = end_loc.charpos + end_len;
          })
 
-let mk_extract_target dst_lang contents rule_ids =
-  let dst_lang =
+(* Compute the rules that should be run for the extracted
+   language.
+
+   Note: for normal targets, this decision is made on the Python
+   side. Implementing that for extracted targets would be far more
+   annoying, however. The main implication of selecting the rules
+   here is, if your target is html and your dest lang is javascript,
+   there is no way to ignore an html path for a specific javascript
+   rule. In general extract mode does not allow you to select rules to
+   run on extracted targets so this is in keeping with that design *)
+let rules_for_extracted_lang (all_rules : Rule.t list) =
+  let rules_for_lang_tbl = Hashtbl.create 10 in
+  let memo xlang =
+    match Hashtbl.find_opt rules_for_lang_tbl xlang with
+    | Some rules_for_lang -> rules_for_lang
+    | None ->
+        let rule_ids_for_lang =
+          all_rules
+          |> List.mapi (fun i r -> (i, r))
+          |> List.filter (fun (_i, r) ->
```

Comment:
`List.filteri` ?

Reply:
No, I need the pair for later when I add it to the table